### ISSUE TITLE
front: stdcm: we sometimes try to add deleted trains to the space time chart

### ISSUE THREAD

**Issue by eckter** [OpenRailAssociation/osrd#8538]
### What happened?

![image](https://github.com/user-attachments/assets/53982555-1c2c-4608-9708-10feeead42a3)

### How can we reproduce it (as minimally and precisely as possible)?

1. Add a train to a timetable
2. Do an STDCM request that would show this train on the space time chart (debug version)
3. Without closing the tab, remove the train from the timetable
4. Without refreshing the stdcm page, try another request

### On which environments the bug occurs?

Local

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

77ed9b9e027aaad67ad2b9a049675f53226e9be8

**Comment by Maymanaf** [comment]
This issue is no longer relevant

**Comment by eckter** [comment]
Sorry but it it still relevant, I could reproduce with the same steps. The only extra step now is to enable the debug elements to have a space-time chart. 

The error now looks like this:

![image](https://github.com/user-attachments/assets/db29c9dd-ecbc-41f6-b73f-a69cb1c5cb90)

**Comment by kmer2016** [comment]
Yes, I can confirm that I was able to reproduce the bug. However, could you clarify what you were expecting @eckter ? It seems that the error message is being returned by the backend.

**Comment by eckter** [comment]
Given the inputs, it's normal that the back-end returns an error.

We try to project a train that was deleted, it can't work out. Either we shouldn't try to project it in the first place, or we should *expect* that an error can happen and "catch" it without displaying anything to the user.

**Comment by RomainValls** [comment]
Can't reproduce with @Yohh and @SarahBellaha   since we don't have the space time chart in debug mode anymore. No error toast.
